src/pages/detail/Detail.js

```diff
@@ -1,7 +1,374 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
+import styled from "styled-components";
+import Modal from "react-modal";
+import ModalOne from "./ModalOne";
+import Slider from "react-slick";
+import "slick-carousel/slick/slick.css";
+import "slick-carousel/slick/slick-theme.css";
 
 const Detail = () => {
-  return <div>Detail</div>;
+  // const [isModal, setIsModal] = useState(false);
+  const [modalIsOpen, setIsOpen] = React.useState(false);
+  // const [closeModal, SetCloseModal] = useState();
+  const [imageIndex, setImageIndex] = useState(0);
+
+  const [bottomPage, setBottomPage] = useState("객실안내");
+  const [room, setRoom] = useState({});
+
+  useEffect(() => {
+    fetch("data/detailData.json", {
+      method: "GET",
+    })
+      .then(res => res.json())
+      .then(data => {
+        setRoom(data);
+        // console.log(data);
+      });
+  }, []);
+
+  // const modalClose = () => {
+  //   setIsModal(!isModal);
+  // };
+
+  // function openModal() {
+  //   setIsOpen(true);
+  // }
+
+  function afterOpenModal() {
+    // references are now sync'd and can be accessed.
+    // subtitle.style.color = "#f00";
+  }
+
+  function closeModal() {
+    setIsOpen(false);
+  }
+
+  let settings = {
+    dots: true,
+    infinite: true,
+    speed: 500,
+    slidesToShow: 3,
+    slidesToScroll: 3,
+    arrows: true,
+    autoplay: false,
+    autoplaySpeed: 5000,
+    centerMod: true,
+  };
+  return (
+    <Container>
+      <Maintop>
+        <AsideImages>
+          <AsideImage
+            src={room.image && room.image[imageIndex].image}
```

Comment:
옵셔널 체이닝을 활용할 수 있습니다.
```suggestion
            src={room.image?.[imageIndex].image}
```

Reply:
참고하겠습니다!